### PR TITLE
Miscellaneous fixes to get the plugin working for more use cases

### PR DESCRIPTION
**Pull request by jasononeil**
* Fix the environment variables being passed through
* Make sure the documentation for which environment variables exist is up to date
* Fix docker compose file being wrong
* A few small fixes for paths of where to find screenshots, and where to save them, and how to load them on the HTML we deploy

### CODE REVIEW

File: gallery/src/gallery.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 class GalleryTemplate < Struct.new(
   :buildkite_pipeline_name,
-  :gallery_url_base,
+  :path_to_gallery_in_s3,
```
Comment (jasononeil) [inline gallery/src/gallery.rb:9]: I just renamed this so it's more obvious what it is

File: gallery/src/gallery.rb [hunk 2]
```diff
@@ -51,9 +51,9 @@ def has_canonical_branch?
   ### Helpers
 
   def images_for_path(path)
-    Dir[File.join(path, '*.png')].map do |filename|
+    Dir["#{path}/**/*.png"].map do |filename|
```
Comment (jasononeil) [inline gallery/src/gallery.rb:54]: I changed the glob path to search recursively

File: gallery/create_gallery.sh [hunk 3]
```diff
@@ -82,17 +82,19 @@ mkdir -p "$triggering_commit_images_dir"
...
-  -e triggering_commit_images_dir -e canonical_branch_images_dir -e gallery_url_base \
+  -e triggering_commit_images_dir -e canonical_branch_images_dir -e path_to_gallery_in_s3 \
   -v "$PWD:/app" -w /app \
   generate-gallery \
   "/app/gallery/src/gallery.rb" "${triggering_commit_images_dir}/${gallery_filename}"
 
 section "Upload screenshots + .html to ${gallery_bucket_name}"
 aws s3 sync --delete --acl public-read \
-  "$triggering_commit_images_dir" "s3://${gallery_bucket_name}/${triggering_repo_and_commit}"
+  "${triggering_commit_images_dir}" "s3://${gallery_bucket_name}${path_to_gallery_in_s3}${triggering_repo_and_commit}"
```
Comment (jasononeil) [inline gallery/create_gallery.sh:97]: Previously we saved things to the root of the bucket, now we allow saving it to a subfolder

File: gallery/create_gallery.sh [hunk 4]
```diff
@@ -82,17 +82,19 @@ mkdir -p "$triggering_commit_images_dir"
 buildkite-agent artifact download "${screenshot_pattern}" "${triggering_commit_images_dir}/"
 
 section "Build screenshot gallery .html"
-docker-compose run \
+docker-compose \
+  --file docker-compose.yml \
```
Comment (jasononeil) [inline gallery/create_gallery.sh:86]: Some pipelines (_cough_ murmur) set a `DOCKER_COMPOSE_FILE` environment variable on the entire buildkite pipeline, and that can mess with this plugin. So I added this to be explicit about which compose file we're looking for.

File: README.md [hunk 5]
```diff
@@ -18,6 +18,7 @@ steps:
       - cultureamp/screenshot-gallery:
           canonical_repo_and_branch: "murmur/master"
           bucket_name: "example-screenshot-gallery"
+          base_url: "http://s3.amazonaws.com/my-bucket-bucket"
```
Comment (sebpearce) [inline README.md:21]: `my-bucket-bucket` is an interesting name

Comment (jasononeil) [reply README.md:21]: Heh, yeah let's change that

Comment (jasononeil) [reply README.md:21]:
```suggestion
          base_url: "http://s3.amazonaws.com/example-screenshot-gallery"
```

File: plugin.yml [hunk 6]
```diff
@@ -5,21 +5,50 @@ requirements:
...
+    # A Github access token with permission to post status results back to commits / pull requests. (Must have the repo:status permission).
+    # Either `github_status_access_token` or `github_status_access_token_param_path` must be supplied.
+    # eg. 123abc456def789ghijklmn123op
+    github_status_access_token:
+      type: string
+    # The path to fetch `github_status_access_token` from Parameter Store (SSM).
+    # Either `github_status_access_token` or `github_status_access_token_param_path` must be supplied.
+    # eg. /my/parameters/GITHUB_STATUS_ACCESS_TOKEN
+    github_status_access_token_param_path:
+      type: stirng
```
Comment (sebpearce) [inline plugin.yml:47]:
```suggestion
      type: string
```

File: gallery/src/gallery.rb (hunk 2 above)
Comment (sebpearce) [reply gallery/src/gallery.rb:54]: good glob

File: hooks/command [hunk 7]
```diff
@@ -6,12 +6,16 @@ set -u
...
   # Pass through require plugin properties:
   export GALLERY_BUCKET_NAME="$BUILDKITE_PLUGIN_SCREENSHOT_GALLERY_BUCKET_NAME"
+  export GALLERY_BASE_URL="$BUILDKITE_PLUGIN_SCREENSHOT_GALLERY_BASE_URL"
   export CANONICAL_REPO_AND_BRANCH="$BUILDKITE_PLUGIN_SCREENSHOT_GALLERY_CANONICAL_REPO_AND_BRANCH"
   export BUILDKITE_GITHUB_ORG="$BUILDKITE_PLUGIN_SCREENSHOT_GALLERY_GITHUB_ORG"
   export BUILDKITE_GITHUB_REPO="$BUILDKITE_PLUGIN_SCREENSHOT_GALLERY_GITHUB_REPO"
 
   # ... and the optional ones:
   export SCREENSHOT_PATTERN="${BUILDKITE_PLUGIN_SCREENSHOT_GALLERY_SCREENSHOT_PATTERN:-}"
+  export PATH_TO_GALLERY_IN_S3="$BUILDKITE_PLUGIN_SCREENSHOT_GALLERY_PATH_TO_GALLERY_IN_S3"
```
Comment (sebpearce) [inline hooks/command:16]:
```suggestion
  export PATH_TO_GALLERY_IN_S3="${BUILDKITE_PLUGIN_SCREENSHOT_GALLERY_PATH_TO_GALLERY_IN_S3:-}"
```

**Review by sebpearce — APPROVED** [review]

**Pull request merged**